### ISSUE TITLE
SlevomatCodingStandard.PHP.UselessParentheses removes also usefull parentheses

### ISSUE THREAD

**Issue by VasekPurchart** [slevomat/coding-standard#425]
`UselessParentheses` [currently](https://github.com/slevomat/coding-standard/blob/4.7.0/tests/Sniffs/PHP/data/uselessParenthesesErrors.php) removes parentheses around really simple expressions, but also for all conditions in ternary expressions. Since these conditions can be more complex (multiple lines, function calls, multiple epressions joined with operators etc.) it can be much less readable without the parentheses (although they are *technically* still not needed).

So I think there should be an option to evaluate these as useful leaving only single "simple" expressions as useless and probably also considering these expressions with unary operators (such as `!`) useless as well.

Fo example:

```php
($foo) // still useless
(!$foo) // still useless
($foo && $bar) // useful
($foo !== $bar) // useful
```

(of course inplace of `$foo` and `$bar` would be more complicated expressions generally)

**Comment by alcaeus** [comment]
One could argue that if your ternary expressions are so complicated that you need parentheses to make them readable, you should probably refactor them into something more readable (e.g. simple method calls that actually tell you what the check is about).

**Comment by VasekPurchart** [comment]
It does not need to be necessary complex as in difficult to understand, but also "visually" complex, which can be even a function call with multiple parameters. Extracting it into a variable or method etc. is always possible, but sometimes not really practical and from readability point of view can make things even worse.

**Comment by carusogabriel** [comment]
Other case that parentheses might be useful:
```php
($foo['bar'] ?? false) ? true : false
```

**Comment by CZechBoY** [comment]
I consider usefull parenthesis in instanceof case.
```php
$abc !== null && !($abc instanceof SomeClass)
```

**Comment by kukulich** [comment]
New option added in https://github.com/slevomat/coding-standard/commit/c0db5807a8134a726466381f9ec81d8bdb6b4e2e

**Comment by Majkl578** [comment]
> I consider usefull parenthesis in instanceof case.

I disagree, those are useless.
